lib/ae_mdw_web/plugs/paginated_plug.ex

```diff
@@ -57,27 +63,32 @@ defmodule AeMdwWeb.Plugs.PaginatedPlug do
       |> assign(:offset, {limit, page})
       |> clean_query()
     else
-      {:error, error_msg} ->
+      {:error, %ErrInput{message: message}} ->
         conn
         |> put_status(:bad_request)
-        |> Controller.json(%{"error" => error_msg})
+        |> Controller.json(%{"error" => message})
         |> halt()
     end
   end
 
   def call(conn, _opts), do: conn
 
-  defp extract_direction_and_scope(%{"range_or_dir" => "forward"}, _txi_scope?, _state),
-    do: {:ok, :forward, @default_scope}
-
-  defp extract_direction_and_scope(%{"range_or_dir" => "backward"}, _txi_scope?, _state),
-    do: {:ok, :backward, @default_scope}
-
-  defp extract_direction_and_scope(%{"range_or_dir" => range} = params, txi_scope?, state) do
-    params
-    |> Map.delete("range_or_dir")
-    |> Map.put("range", range)
-    |> extract_direction_and_scope(txi_scope?, state)
+  defp extract_direction_and_scope(
+         %{"scope" => "transaction:" <> transaction_scope} = params,
+         _txi_scope,
+         state
+       ) do
+    with tx_hashes when length(tx_hashes) > 0 and length(tx_hashes) <= 2 <-
```

Comment:
an alternative suggest would be to pattern match with `[_hash1, _hash2] = tx_hashes`, similar, just another flavour

Reply:
I can't pattern-match on this since it's valid for it to be a single element array too

Reply:
I mean this format `[_hash1 | _hash2] = tx_hashes` for a more compact condition but it's fine the lengthy one (:

Reply:
nvm, it would be lengthier checking the rest of the list